### ISSUE TITLE
PHP 8.1.0-dev compat: "Passing null to parameter ... is deprecated"

### ISSUE THREAD

**Issue by morrisonlevi** [composer/composer#9770]
Using PHP's master branch:
```bash
$ php --version
PHP 8.1.0-dev (cli) (built: Mar 11 2021 11:40:12) ( NTS DEBUG )
Copyright (c) The PHP Group
Zend Engine v4.1.0-dev, Copyright (c) Zend Technologies
```

Today's composer:
```bash
$  php ../composer.phar --version
Composer version 2.0.11 2021-02-24 14:57:23
```

Command:
```
$ php ../composer.phar create-project --ignore-platform-reqs symfony/symfony-demo app
```
Output:
```
Creating a "symfony/symfony-demo" project at "./app"
Deprecation Notice: preg_match(): Passing null to parameter #2 ($subject) of type string is deprecated in phar:///Users/levi.morrison/Projects/symfony-demo/composer.phar/src/Composer/Command/CreateProjectCommand.php:346
Deprecation Notice: preg_replace(): Passing null to parameter #3 ($subject) of type array|string is deprecated in phar:///Users/levi.morrison/Projects/symfony-demo/composer.phar/vendor/composer/semver/src/VersionParser.php:53
Installing symfony/symfony-demo (v1.0.5)
  - Downloading symfony/symfony-demo (v1.0.5)
  - Installing symfony/symfony-demo (v1.0.5): Extracting archive
Created project in /Users/levi.morrison/Projects/symfony-demo/build/app
Deprecation Notice: strlen(): Passing null to parameter #1 ($string) of type string is deprecated in phar:///Users/levi.morrison/Projects/symfony-demo/composer.phar/vendor/justinrainbow/json-schema/src/JsonSchema/Constraints/Constraint.php:48
...
```

As far as I can tell from a quick look, it doesn't parse the PHP version correctly and so it's `null`.

**Comment by Seldaek** [comment]
Duplicate of https://github.com/composer/composer/issues/9718

**Comment by Seldaek** [comment]
Ah sorry this may not be the exact same issue, that one may actually be fixable sooner than 2.2 so reopening.

**Comment by Seldaek** [comment]
Can you check again with the latest version? I believe it's fixed but don't have a 8.1 at hand to try.

**Comment by morrisonlevi** [comment]
All of the deprecation notices that came directly from composer are fixed when running this command:

```bash
php composer.phar create-project --ignore-platform-reqs symfony/symfony-demo app
```

```bash
php composer.phar --version
```
> Composer version 2.0.12 2021-04-01 10:14:59

There is still this deprecation notice in something that composer vendors in, though:

```
Deprecation Notice: strlen(): Passing null to parameter #1 ($string) of type string is deprecated in phar:///build/php/master/composer.phar/vendor/justinrainbow/json-schema/src/JsonSchema/Constraints/Constraint.php:48
```

**Comment by Seldaek** [comment]
Yes we're waiting on https://github.com/justinrainbow/json-schema/pull/657 for this.. Not so urgent though given the 8.1 timeline :)

Alright then closing this.

**Comment by Raghav9888** [comment]
[https://prnt.sc/EgCD4omoM-Gn](url)

I can't resole this error please help me
